### PR TITLE
OESS-168:Remove clang warnings.

### PR DESCRIPTION
**Pull request by hyoklee**
I removed some clang warnings.

### CODE REVIEW

File: hl/src/H5LTanalyze.c [hunk 1]
```diff
@@ -1947,7 +1947,7 @@ static int yy_get_next_buffer (void)
 					yyrestart( yyin );
 
 					/*FALLTHROUGH*/
-
+                                        __attribute__((fallthrough));
```
Comment (byrnHDF) [inline hl/src/H5LTanalyze.c]: This needs to be changed in the source file - this file is generated.

**Review by byrnHDF — CHANGES_REQUESTED** [review]

File: hl/test/test_ds.c [hunk 2]
```diff
@@ -5210,6 +5210,8 @@ test_attach_detach(void)
 
     HL_TESTING2("permutations of attaching and detaching");
 
+    gid = var1_id = var2_id = var3_id = -1;
```
Comment (byrnHDF) [inline hl/test/test_ds.c]: We should be encouraging that -1 should be H5I_INVALID_HID for hid_t variables.

**Review by byrnHDF — CHANGES_REQUESTED** [review]

**Review by qkoziol — APPROVED** [review]

**Review by byrnHDF — APPROVED** [review]

File: hl/test/test_packet.c [hunk 3]
```diff
@@ -45,10 +45,10 @@ typedef struct particle_t {
  * a static array of particles for writing and checking reads
  *-------------------------------------------------------------------------
  */
-static particle_t testPart[NRECORDS] = {{"zero", 0, 0, 0.0f, 0.0f},    {"one", 10, 10, 1.0f, 10.0f},
-                                        {"two", 20, 20, 2.0f, 20.0f},  {"three", 30, 30, 3.0f, 30.0f},
-                                        {"four", 40, 40, 4.0f, 40.0f}, {"five", 50, 50, 5.0f, 50.0f},
-                                        {"six", 60, 60, 6.0f, 60.0f},  {"seven", 70, 70, 7.0f, 70.0f}};
+static particle_t testPart[NRECORDS] = {{"zero", 0, 0, 0.0f, 0.0},    {"one", 10, 10, 1.0f, 10.0},
```
Comment (byrnHDF) [inline hl/test/test_packet.c:48]: Just noticed the "f" is still on 4th value in the arrays - shouldn't it match the change to the 5th value?

**Review by byrnHDF — COMMENTED** [review]
Better - just one new question.

File: hl/test/test_packet.c (hunk 3 above)
Comment (hyoklee) [reply hl/test/test_packet.c:48]: 4th value is for float testing. 5th value is for double testing. clang doesn't complain for 4th value.

**Pull request merged**